PyPDF2/_writer.py

```diff
@@ -871,70 +856,94 @@ def _sweep_indirect_references(
-                    raise ValueError(
-                        f"I/O operation on closed file: {data.pdf.stream.name}"
-                    )
-                newobj = (
-                    extern_map.get(data.pdf, {})
-                    .get(data.generation, {})
-                    .get(data.idnum, None)
-                )
-                if newobj is None:
-                    try:
-                        newobj = data.pdf.get_object(data)
-                        self._objects.append(None)  # placeholder
-                        idnum = len(self._objects)
-                        newobj_ido = IndirectObject(idnum, 0, self)
-                        if data.pdf not in extern_map:
-                            extern_map[data.pdf] = {}
-                        if data.generation not in extern_map[data.pdf]:
-                            extern_map[data.pdf][data.generation] = {}
-                        extern_map[data.pdf][data.generation][data.idnum] = newobj_ido
-                        newobj = self._sweep_indirect_references(extern_map, newobj)
-                        self._objects[idnum - 1] = newobj
-                        return newobj_ido
-                    except (ValueError, RecursionError):
-                        # Unable to resolve the Object, returning NullObject instead.
-                        warnings.warn(
-                            f"Unable to resolve [{data.__class__.__name__}: {data}], "
-                            "returning NullObject instead",
-                            PdfReadWarning,
-                        )
-                        return NullObject()
-                return newobj
+                    data = self._resolve_indirect_object(self._add_object(data))
+
+                update_hashes = []
+
+                # Data changed what mead hash value changed
+                if parent[key_or_id] != data:
+                    update_hashes = [
+                        parent.hash_value()
+                    ] + [
+                        grant_parent.hash_value()
+                        for grant_parent in grant_parents
+                    ]
+                    parent[key_or_id] = data
+
+                # Update old hash value to new hash value
+                for old_hash in update_hashes:
+                    indirect_ref = self._idnum_hash.pop(old_hash, None)
+
+                    if indirect_ref is not None:
+                        indirect_ref_obj = indirect_ref.get_object()
+
+                        if indirect_ref_obj is not None:
+                            self._idnum_hash[indirect_ref_obj.hash_value()] = indirect_ref
+            # while len(stack)
+
+    def _resolve_indirect_object(self, data: IndirectObject) -> IndirectObject:
+        """
+        Resolves indirect object to this pdf indirect objects.
+
+        If it is a new object then it is added to self._objects and new idnum is given and
+        generation is always 0.
```

Comment:
This just prevents a line break in the GitHub diff view:

```suggestion
        If it is a new object then it is added to self._objects
        and new idnum is given and generation is always 0.
```